pkg/collector/corechecks/telemetry/check_test.go

```diff
@@ -8,31 +8,37 @@ package telemetry
 import (
 	"testing"
 
-	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/require"
 
 	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	telemetryComponent "github.com/DataDog/datadog-agent/comp/core/telemetry"
+	"github.com/DataDog/datadog-agent/comp/core/telemetry/telemetryimpl"
 	"github.com/DataDog/datadog-agent/pkg/aggregator/mocksender"
 	"github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
 func TestCheck(t *testing.T) {
-	reg := prometheus.NewRegistry()
+	telemetryMock := fxutil.Test[telemetryComponent.Mock](t, telemetryimpl.MockModule())
+	telemetryMock.Reset()
 
 	func() {
-		gauge := prometheus.NewGaugeVec(prometheus.GaugeOpts{Subsystem: "test", Name: "_gauge"}, []string{"foo"})
-		gauge.WithLabelValues("bar").Set(1.0)
-		gauge.WithLabelValues("baz").Set(2.0)
-		reg.MustRegister(gauge)
+		gauge := telemetryMock.NewGaugeWithOpts("test", "gauge", []string{"foo"}, "", telemetryComponent.Options{
+			DefaultMetric: true,
+		})
 
-		count := prometheus.NewCounter(prometheus.CounterOpts{Subsystem: "test", Name: "_counter"})
+		gauge.WithTags(map[string]string{"foo": "bar"}).Set(1.0)
+		gauge.WithTags(map[string]string{"foo": "baz"}).Set(2.0)
+
+		count := telemetryMock.NewCounterWithOpts("test", "counter", []string{}, "", telemetryComponent.Options{
+			DefaultMetric: true,
+		})
 		count.Add(4.0)
```

Comment:
❓ I do not see where you use the count variable after the add, how does it works ?

Reply:
I don't use. 

This function create telemetry metrics. Everytime we create telemetry metrics they get store in a Prometheus registry https://github.com/DataDog/datadog-agent/blob/0c032de1d8e762dc0e3f7aab885ab0ad4297215a/comp/core/telemetry/telemetryimpl/telemetry.go#L264-L265

Later in the test, we assert on the sender manager mock that the right function is called with the right argument values. 
https://github.com/DataDog/datadog-agent/blob/11f5a7e468088eb0a89be0cfec8be59ed43e4c83/pkg/collector/corechecks/telemetry/check_test.go#L47